app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # Guest user (not logged in)
+
+    # Post abilities
+    can :destroy, Post, user_id: user.id
+    can :destroy, Post, user: { role: 'admin' }
```

Comment:
- [ ] You did great work 😃, however, as mentioned by the previous reviewer, an admin user should be able to see the delete button for the post, normally, we are not supposed to delete other's posts or comments, is only the owner of the post who is suppose to see the delete button. Let's take users `A` and `B`, when `A` add a post, `A` can delete or update it, but not `B`, also if `B` add a post, `A` can't delete or update it. Kindly, fix it 😃

Reply:
Thank you for your insight! I didn't notice that, The buttons visibility are working properly now. 😄